tracer/src/Datadog.Trace/Logging/Internal/DatadogLoggingFactory.cs

```diff
@@ -163,13 +164,19 @@ private static string GetDefaultLogDirectory(string? logDirectory)
 #if NETFRAMEWORK
             logDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
 #else
-            if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            var isWindows = System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows);
+
+            if (ServerlessMiniAgent.GetIsAzureFunction())
```

Comment:
using `Serverless.IsAzureFunction` instead (which is initialized `usingGetIsAzureFunction()` always evaluates to false, but calling the function directly works.

I suspect this is due to the way those fields of `Serverless` are initialized, if anyone has some insights please let me know!

Reply:
This is due to the way the property was declared:
```csharp
internal static bool IsAzureFunction { get; private set; } = GetIsAzureFunction();
```

This declares a property with a private setting, and sets the property's value to ` GetIsAzureFunction()`. It is basically shorthand for:


```charp
internal class ServerlessMiniAgent
{
    internal static bool IsAzureFunction { get; private set; } = GetIsAzureFunction();

    // note this happens in the static constructor because IsAzureFunction is static,
    // this code only runs once, the first time you touch the ServerlessMiniAgent type,
    static ServerlessMiniAgent()
    {
        IsAzureFunction = GetIsAzureFunction();
    }
```

Reply:
I find that having both `IsAzureFunction` and `GetIsAzureFunction()` is confusing. One computed the value on every call, while the other one caches the value and always returns the same, and it's not clear which one is which.

Unless you have a specific reason to cache the value (e.g. performance in the hot path), maybe think of removing `IsAzureFunction` entirely and always calling `GetIsAzureFunction()`.